modules/images/webp-uploads/load.php

```diff
@@ -154,6 +180,24 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 				continue;
 			}
 
+			$original_thumbnail_image_filesize = isset( $properties['filesize'] ) ? (int) $properties['filesize'] : 0;
+			$webp_thumbnail_image_filesize     = isset( $source['filesize'] ) ? (int) $source['filesize'] : 0;
+			if ( $original_thumbnail_image_filesize > 0 && $webp_thumbnail_image_filesize > 0 ) {
+				/** This filter is documented in modules/images/webp-uploads/load.php */
+				if (
+					apply_filters( 'webp_uploads_discard_larger_generated_images', true )
+					&& $webp_thumbnail_image_filesize >= $original_thumbnail_image_filesize
+				) {
```

Comment:
This is exactly the same logic as in https://github.com/WordPress/performance/pull/418/files#diff-71ed188a16b79d179868ee80e76f775462a451e871fc3d6214fcacca3edcea8cR99-R115 I think. Let's implement a separate function for this, especially for the filter.

I would suggest implementing a `webp_uploads_should_discard_additional_image_file( array $original, array $additional, string $size_name ) : bool` function. That function could include all this logic, comparing the file sizes if they are not `0` and then run the filter, and return the boolean result.

We could then call this function in both places instead of duplicating all this logic.

Reply:
Thanks for the suggestion, Felix.

Do we need `string $size_name` parameter in the function as we don't check the image size name in our logic? or do we need to check any logic on the image size name?

Reply:
I have updated PR without `string $size_name`.

Reply:
Thank you @mukeshpanchal27, that makes sense. I was probably thinking to include `$size_name` in the filter as an additional parameter, but that shouldn't be necessary for now. We can always add it later if it comes up as a request.